pkg/occlient/occlient.go

```diff
@@ -3287,6 +3287,14 @@ func injectS2IPaths(existingVars []corev1.EnvVar, s2iPaths S2IPaths) []corev1.En
 
 }
 
+// IsDeploymentConfigSupported checks if DeploymentConfig type is present on the cluster
+func (c *Client) IsDeploymentConfigSupported() (bool, error) {
+	const Group = "apps.openshift.io"
+	const Version = "v1"
+
+	return c.isResourceSupported(Group, Version, "DeploymentConfig")
```

Comment:
This seems to panic for me, when run against a s2i component in experimental mode, as the client is nil
```

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1c29be0]

goroutine 1 [running]:
github.com/openshift/odo/pkg/occlient.(*Client).isResourceSupported(0x0, 0x24f97a2, 0x11, 0x24ea377, 0x2, 0x24f7688, 0x10, 0xc000261300, 0x0, 0x0)
        /home/mrinaldas/go/src/github.com/openshift/odo/pkg/occlient/occlient.go:3341 +0xf0
github.com/openshift/odo/pkg/occlient.(*Client).IsDeploymentConfigSupported(0x0, 0x0, 0x0, 0x0)
        /home/mrinaldas/go/src/github.com/openshift/odo/pkg/occlient/occlient.go:3295 +0x8c
github.com/openshift/odo/pkg/odo/cli/component.(*ListOptions).Complete(0xc0003ceac0, 0x24eafa4, 0x4, 0xc0006a3680, 0x39712d8, 0x0, 0x0, 0x0, 0x0)
```

Reply:
Even fails for devfile components.

Reply:
resolved, so we cannot create the `Client` only if the resource exists as thats a catch 22 situation.